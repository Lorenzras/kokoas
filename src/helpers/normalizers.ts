import { isField } from './utils';
import { ContactField, CustomerForm, InputField } from './../types/forms';


const convertContactsObj = (contactsObj : ContactField) : Omit<CustomerTypes.Data['contacts'], 'type'> => {
  return {
    
    value: Object.values(contactsObj).map(({ classification, value, label })=>{
      return {
        id: '',
        value: {
          contactType: { value: label },
          contactValue: { value },
          classification: {
            value: classification.value,
          },
        },
      };
    }),
  };
};

export const convertCustFormState = (state: CustomerForm) : Partial<CustomerTypes.SavedData>[] => {
  const kintoneRecord = state.customers.map((cust) => {

    return Object.entries(cust).reduce((prev, curr) => {
      const [fieldName, value] = curr;
      if (isField(value)){
        return { ...prev, [fieldName]: { value: (value as InputField).value } };
      } else if (fieldName === 'contacts'){

        return { ...prev, [fieldName] : convertContactsObj(value as ContactField) };
      }
      return prev;
    }, {});

  });



  return kintoneRecord;
};

export const testFunc = () => {

  return 'hello';
};