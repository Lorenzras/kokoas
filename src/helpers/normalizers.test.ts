import { convertCustFormState, testFunc } from './normalizers';
import { custFormStateTestData } from './test/testData';

describe('Converters', ()=> {
  test('is hello', ()=> {
    expect(testFunc()).toBe('hello');
  });
  
  test('returns a kintone record', ()=>{
    const fn = convertCustFormState(custFormStateTestData());
    expect(fn[0]).toMatchSnapshot({
      contacts: expect.any(Object),
    });
  });
});
