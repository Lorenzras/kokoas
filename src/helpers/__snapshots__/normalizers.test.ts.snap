// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Converters returns a kintone record 1`] = `
Object {
  "address1": Object {
    "value": "テスト",
  },
  "address2": Object {
    "value": "どこでも",
  },
  "birthDay": Object {
    "value": "31",
  },
  "birthMonth": Object {
    "value": "10",
  },
  "birthYear": Object {
    "value": "2000",
  },
  "contacts": Any<Object>,
  "fullName": Object {
    "value": "Ras, Lorenz T",
  },
  "fullNameReading": Object {
    "value": "ラス　ロレンズ　トンボ",
  },
  "gender": Object {
    "value": "男性",
  },
  "postalCode": Object {
    "value": "441-8124",
  },
}
`;
